### PR TITLE
Show Platform Environment

### PR DESCRIPTION
**Pull request by DhanaDell**
sonic# show platform environment 
                   
                   
Fan Trays:          
   Fan Tray 1:         
       Fan State:                    Normal              
       Fan1 Speed:                   12300 RPM           
       Fan2 Speed:                   14850 RPM           
   Fan Tray 2:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   15000 RPM           
   Fan Tray 3:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   14850 RPM           
   Fan Tray 4:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   14700 RPM           
   Fan Tray 5:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   14850 RPM           
   Fan Tray 6:         
       Fan State:                    Normal              
--more--
       Fan1 Speed:                   12450 RPM           
       Fan2 Speed:                   14700 RPM           
   Fan Tray 7:         
       Fan State:                    Normal              
       Fan1 Speed:                   12600 RPM           
       Fan2 Speed:                   14700 RPM           
                   
                   
Onboard Temperature Sensors:
   Fan U52:                         25 degrees C
       Baseboard U3:                 25 degrees C        
       Fan U17:                      21 degrees C        
       Near CPU:                     51 degrees C        
       PSU1 hotspot:                 30 degrees C        
       PSU1 inlet:                   22 degrees C        
       PSU2 hotspot:                 31 degrees C        
       PSU2 inlet:                   23 degrees C        
       SW U04:                       28 degrees C        
       SW U14:                       22 degrees C        
       SW U16:                       21 degrees C        
       SW U4403:                     40 degrees C        
       SW U52:                       21 degrees C        
       SW interal:                   43 degrees C        
                   
                   
--more--
PSUs:               
   PSU1:               
       FAN RPM:                      15800 RPM           
       Hotspot Temperature:          30 degrees C        
       Inlet Temperature:            22 degrees C        
       Input Current:                1.04 Amps           
       Input Power:                  210 Watts           
       Input Voltage:                207.90 Volts        
       Output Current:               16.50 Amps          
       Output Power:                 198 Watts           
       Output Voltage:               12 Volts            
   PSU2:               
       FAN RPM:                      15800 RPM           
       Hotspot Temperature:          31 degrees C        
       Inlet Temperature:            23 degrees C        
       Input Current:                0.96 Amps           
       Input Power:                  192 Watts           
       Input Voltage:                210.10 Volts        
       Output Current:               15 Amps             
       Output Power:                 180 Watts           
       Output Voltage:               12 Volts            
                   
                   
coretemp-isa-0000   
   Adapter: ISA adapter
--more--
       Core 0:                       +52.0 C  (high = +82.0 C, crit = +104.0 C)
       Core 1:                       +52.0 C  (high = +82.0 C, crit = +104.0 C)
       Physical id 0:                +52.0 C  (high = +82.0 C, crit = +104.0 C)

### CODE REVIEW

File: src/CLI/clitree/cli-xml/platform.xml [hunk 1]
```diff
@@ -1,50 +1,28 @@
...
-
-   <COMMAND
-         name="show platform"
-         help="Show platform information"
-     >
-         <ACTION>
-        python $SONIC_CLI_ROOT/sonic-cli-pfm.py get_openconfig_platform_components platform_show.j2 ${__full_line}
-        </ACTION>
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE CLISH_MODULE [
```
Comment (joyas-joseph) [inline src/CLI/clitree/cli-xml/platform.xml]: Looks like you are committing the processed file.

File: src/CLI/actioner/sonic-cli-pfm.py [hunk 2]
```diff
@@ -21,108 +21,25 @@
...
-                     errDict = {}
-                     for dict in errList:
-                         for k, v in dict.iteritems():
-                              errDict[k] = v
+    path = cc.Path('/restconf/data/openconfig-platform:components/component=%s'%args[0])
+    response = aa.get(path)
+    if response.ok():
+        show_cli_output(sys.argv[3], response.content)
+    else:
+        print "Failed"
```
Comment (joyas-joseph) [inline src/CLI/actioner/sonic-cli-pfm.py]: Use response.error_message() for the failure case.

File: models/yang/openconfig-platform-types.yang [hunk 3]
```diff
@@ -0,0 +1,347 @@
+module openconfig-platform-types {
```
Comment (joyas-joseph) [inline models/yang/openconfig-platform-types.yang]: This yang should be in the common directory itself.

File: src/CLI/actioner/sonic-cli-pfm.py (hunk 2 above)
Comment (joyas-joseph) [inline src/CLI/actioner/sonic-cli-pfm.py]: Lets not print just "Failed". Let's use the format "%Error: <message>". response.error_message() would give the reason.

File: src/CLI/clitree/cli-xml/platform.xml (hunk 1 above)
Comment (joyas-joseph) [reply src/CLI/clitree/cli-xml/platform.xml]: This is the processed file. Please revert and commit changes to the original

**Comment by DhanaDell** [comment]
Output -
~~~text
root@sonic:/# /usr/sbin/cli/clish_start 
sonic# show platform environment 
                   
                   
Fan Trays:          
   Fan Tray 1:         
       Fan State:                    Normal              
       Fan1 Speed:                   12300 RPM           
       Fan2 Speed:                   14850 RPM           
   Fan Tray 2:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   15000 RPM           
   Fan Tray 3:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   14850 RPM           
   Fan Tray 4:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   14700 RPM           
   Fan Tray 5:         
       Fan State:                    Normal              
       Fan1 Speed:                   12150 RPM           
       Fan2 Speed:                   14850 RPM           
   Fan Tray 6:         
       Fan State:                    Normal              
--more--
       Fan1 Speed:                   12450 RPM           
       Fan2 Speed:                   14700 RPM           
   Fan Tray 7:         
       Fan State:                    Normal              
       Fan1 Speed:                   12600 RPM           
       Fan2 Speed:                   14700 RPM           
                   
                   
Onboard Temperature Sensors:
   Fan U52:                         25 degrees C
       Baseboard U3:                 25 degrees C        
       Fan U17:                      21 degrees C        
       Near CPU:                     51 degrees C        
       PSU1 hotspot:                 30 degrees C        
       PSU1 inlet:                   22 degrees C        
       PSU2 hotspot:                 31 degrees C        
       PSU2 inlet:                   23 degrees C        
       SW U04:                       28 degrees C        
       SW U14:                       22 degrees C        
       SW U16:                       21 degrees C        
       SW U4403:                     40 degrees C        
       SW U52:                       21 degrees C        
       SW interal:                   43 degrees C        
                   
                   
--more--
PSUs:               
   PSU1:               
       FAN RPM:                      15800 RPM           
       Hotspot Temperature:          30 degrees C        
       Inlet Temperature:            22 degrees C        
       Input Current:                1.04 Amps           
       Input Power:                  210 Watts           
       Input Voltage:                207.90 Volts        
       Output Current:               16.50 Amps          
       Output Power:                 198 Watts           
       Output Voltage:               12 Volts            
   PSU2:               
       FAN RPM:                      15800 RPM           
       Hotspot Temperature:          31 degrees C        
       Inlet Temperature:            23 degrees C        
       Input Current:                0.96 Amps           
       Input Power:                  192 Watts           
       Input Voltage:                210.10 Volts        
       Output Current:               15 Amps             
       Output Power:                 180 Watts           
       Output Voltage:               12 Volts            
                   
                   
coretemp-isa-0000   
   Adapter: ISA adapter
--more--
       Core 0:                       +52.0 C  (high = +82.0 C, crit = +104.0 C)
       Core 1:                       +52.0 C  (high = +82.0 C, crit = +104.0 C)
       Physical id 0:                +52.0 C  (high = +82.0 C, crit = +104.0 C)
                   
                   
~~~

**Review by joyas-joseph — APPROVED** [review]

**Pull request merged**